### PR TITLE
Support setting more properties in the JMX Metrics properties file

### PR DESCRIPTION
**Pull request by dehaansa**
**Description:**
Enhance properties file support in JMX Metrics Gatherer as collector-contrib is moving towards using the properties file for communication to JMX MG.

**Existing Issue(s):** https://github.com/open-telemetry/opentelemetry-java-contrib/issues/322

**Testing:**
Added unit tests to validate properties file support & expected interaction between properties file values & command line arguments.

### CODE REVIEW

File: jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java [hunk 1]
```diff
@@ -32,6 +32,16 @@ class JmxConfig {
   static final String JMX_REMOTE_PROFILE = PREFIX + "jmx.remote.profile";
   static final String JMX_REALM = PREFIX + "jmx.realm";
 
+  static final List<String> JAVA_SYSTEM_PROPERTIES =
+      Arrays.asList(
+          "otel.resource.attributes",
```
Comment (mateuszrzeszutek) [inline jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java:37]: About `otel.resource.attributes`: take a look at https://github.com/open-telemetry/opentelemetry-java-contrib/blob/main/jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/GroovyMetricEnvironment.java#L63-L85 -- according to that code, the resource property is already being propagated to the OTel SDK autoconfiguration. Does this not work?

File: jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java [hunk 2]
```diff
@@ -32,6 +32,16 @@ class JmxConfig {
...
+          "javax.net.ssl.keyStore",
+          "javax.net.ssl.keyStorePassword",
+          "javax.net.ssl.keyStoreType",
+          "javax.net.ssl.trustStore",
+          "javax.net.ssl.trustStorePassword",
+          "javax.net.ssl.trustStoreType");
```
Comment (mateuszrzeszutek) [inline jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java:43]: What is consuming these properties? If it's the metric exporter, should these be copied in the `GroovyMetricEnvironment` together with all `otel.*` props?

Comment (mateuszrzeszutek) [inline jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java:43]: What is consuming these properties? If it's the metric exporter, should these be copied in the `GroovyMetricEnvironment` together with all `otel.*` props?

**Comment by mateuszrzeszutek** [comment]
Pinging @rmfitzpatrick @Mrod1598

File: jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java (hunk 1 above)
Comment (dehaansa) [reply jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java:37]: I had missed this component of the groovy metric environment startup! Thanks for pointing it out, I'll have to remove the otel.resource.attributes from the list that gets relocated into system.properties.

File: jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java (hunk 2 above)
Comment (dehaansa) [reply jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/JmxConfig.java:43]: These properties are intended to be used internally to the JMXClient, not the metrics exporter.

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**